src/ChunkedStreamDecoder.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace React\HttpClient;
+
+use Evenement\EventEmitterTrait;
+use Exception;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\Util;
+use React\Stream\WritableStreamInterface;
+
+class ChunkedStreamDecoder implements ReadableStreamInterface
```

Comment:
Should we mark this as `@internal` or should consumers of this library be allowed to rely on this?

Afaict this should only be used internally and should not be part of our public API.